### PR TITLE
Prototype of clipping masks

### PR DESCRIPTION
**Pull request by dvoytenko**
Migrated from https://github.com/ampproject/amp-wp/pull/4062

### CODE REVIEW

File: assets/src/edit-story/masks/index.js [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import StoryPropTypes from '../types';
+
+// Important! This file cannot use `styled-components` or any stateful/context
```
Comment (dvoytenko) [inline assets/src/edit-story/masks/index.js:32]: @swissspidy can you help here? Can we construct a eslint rule to prohibit most of imports here. Specifically we need to disallow `styled-components` anywhere in this folder.

**Review by wassgha — APPROVED** [review]
Looks great!

File: assets/src/edit-story/masks/index.js (hunk 1 above)
Comment (swissspidy) [reply assets/src/edit-story/masks/index.js:32]: You can add an `.eslintrc` file in this folder with the following contents:

```json
{
  "rules": {
    "no-restricted-imports": ["error", { "paths": ["styled-components"] }]
  }
}
```

Can amend the list with `@wordpress/element` (React) and others if needed.

Comment (dvoytenko) [reply assets/src/edit-story/masks/index.js:32]: Awesome! Thanks!

**Pull request merged**